### PR TITLE
build: Upgrade to ForgeGradle 6 and switch to Sugarcane mappings

### PR DESCRIPTION
**Pull request by Jonathing**
Port of The-Aether-Team/The-Aether#1589, The-Aether-Team/The-Aether#1573, and The-Aether-Team/The-Aether#1148 to Nitrogen. Read each individual PR for details.

New points of note
- The buildscript has been modified to remove functionality with Forge's Jar-in-Jar system.
- The buildscript has been copied over from The-Aether-Team/The-Aether@abb6fe213f31d82d726a628eb35d34a6c1c461fe.
- All commits involving modifications to `.circleci` have been cherry-picked from the original PR. I have marked @bconlon1 as the commit author for the cherry-picked commit.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,89 +1,95 @@
...
         }
 
         data {
-            workingDirectory project.file('run')
-
-            //property 'forge.logging.markers', 'SCAN,REGISTRIES,REGISTRYDUMP'
-            //property 'forge.logging.console.level', 'debug'
-
-            args '--mod', project.mod_id, '--all', '--output', file('src/generated/resources/'), '--existing', file('src/main/resources/'), '--existing', file('src/generated/resources/')
+            args '--mod', mod_id, '--all', '--output', file('src/generated/resources/'), '--existing', file('src/main/resources/'), '--existing', file('src/generated/resources/'), '--existing-mod', 'aether_nitrogen'
```
Comment (bconlon1) [inline build.gradle]: Shouldn't include --existing-mod

File: build.gradle [hunk 2]
```diff
@@ -117,37 +123,31 @@ jar {
...
             artifact jar
-            artifact sourcesJar
+            artifact sourceJar
         }
     }
     repositories {
         maven {
             name = "GitHubPackages"
-            url = "https://maven.pkg.github.com/The-Aether-Team/Nitrogen"
+            url = "https://maven.pkg.github.com/The-Aether-Team/The-Aether"
```
Comment (bconlon1) [inline build.gradle]: should reference Nitrogen not The-Aether

**Review by bconlon1 — CHANGES_REQUESTED** [review]

**Review by bconlon1 — APPROVED** [review]
lgtm

**Comment by Jonathing** [comment]
😎👍

**Pull request merged**